module/hotspot/hsperfdata/hsperfdata.go

```diff
@@ -89,59 +162,67 @@ func (m *MetricSet) buildMapStr(entries []PerfDataEntry) common.MapStr {
 	return event
 }
 
-func (m *MetricSet) publishAll() (common.MapStr, error) {
-	f, err := os.Open(m.hsPerfDataPath)
+func (p *ProcStats) publishAll() (common.MapStr, error) {
+	f, err := os.Open(p.hsPerfDataPath)
 	if err != nil {
 		return nil, err
```

Comment:
Thanks for the feedback!

I realized there is another (related) issue. Let's say a non-admin user runs hsbeat and there are multiple Java processes running, from different users. With the current logic, the Fetch loop will return an error when it hits the first file for which the user doesn't have permissions. In that case we wouldn't return any event, making the new feature useless for non-root users (when processes from other users are running).

I'm going to update the PR with a new commit that accumulates errors and tries to return all the successful events (those for the processes the user has permissions on). Only if there are no events it will return the error. If there are successful events, it will just log the errors (this would be the normal case for non-admin users).

I also implemented your suggestion to log the error when opening the file.

I didn't add any user specific treatment (i.e. root vs non-root) since it is not trivial to know whether a user will have permissions on a file (root will always have but non-root may (or may not) have access to files owned by other users. Instead I log debug messages as you suggested.